src/client/Main.cpp

```diff
@@ -1,13 +1,60 @@
+/*
+** EPITECH PROJECT, 2024
+** RType
+** File description:
+** Main.cpp
+*/
+
 #include <SFML/Graphics.hpp>
 #include "Window.hpp"
+#include "network/RTypeClient.hpp"
+#include "GameEngine/GameEngine.hpp"
+
+void updateComponent(size_t id, std::string name, std::vector<uint8_t> data, RTypeClient &conn)
+{
+    std::vector<uint8_t> updateOperation;
+
+    updateOperation.resize(1 + 2 + 2 + data.size());
+    updateOperation[0] = 0x3;
+    updateOperation[1] = (uint16_t)id;
+    for (const auto &compId : conn.getCompNames()) {
+        if (compId.second == name) {
+            updateOperation[3] = (uint16_t)compId.first;
+        }
+    }
```

Comment:
Maybe add a `break;` int the if to stop the loop.
I put this comment even though it is my part to not forget.